2021/day02/solutions/avokado.zig

```diff
@@ -0,0 +1,107 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+
+const Unit = i32;
+
+const Position = struct {
+    horizontal: Unit = 0,
+    depth: Unit = 0,
+};
+
+const ParseError = error {
+    InvalidCharacter,
+};
+
+pub fn main() anyerror!void {
+    const out = std.io.getStdOut().writer();
+
+    var arena_alloc = std.heap.ArenaAllocator.init(std.heap.page_allocator);
+    defer arena_alloc.deinit();
+
+    var allocator = arena_alloc.allocator();
+
+    const in_buffer = blk: {
+        const std_in = std.io.getStdIn();
+        const in_stat = try std_in.stat();
+        break :blk try std_in.readToEndAllocOptions(
+            allocator, 
+            std.math.maxInt(usize), 
+            in_stat.size, 
+            @alignOf(u8), 
+            null
+        );
+    };
+
+    const p1 = try part1(in_buffer);
+    try out.print("{d}\n", .{p1});
+    
+    const p2 = try part2(in_buffer);
+    try out.print("{d}\n", .{p2});
+}
+
+inline fn part1(file_content: []const u8) !i32 {
+    var position = Position{};
+
+    var file_iter = std.mem.split(u8, file_content, "\n");
+    while(file_iter.next()) |line| {
+        if (line.len == 0) { continue; }
+
+        switch (line[0]) {
+            'f' => position.horizontal += try std.fmt.parseInt(Unit, line["forward ".len..], 10),
```

Comment:
Nice simplification just checking the first letter👍

Reply:
Cheers!